code/modules/mob/living/inhand_holder.dm

```diff
@@ -0,0 +1,123 @@
+	..()
+	if(isturf(loc))//don't release on soft-drops
+		release()
+
+/obj/item/clothing/head/mob_holder/proc/release(del_on_release = TRUE)
+	if(isliving(loc))
+		var/mob/living/L = loc
+		L.dropItemToGround(src)
+	if(held_mob)
+		var/mob/living/m = held_mob
+		m.forceMove(get_turf(m))
+		m.reset_perspective()
+		m.setDir(SOUTH)
+		held_mob = null
+	if(del_on_release)
+		qdel(src)
+
+/obj/item/clothing/head/mob_holder/relaymove(mob/user)
+	to_chat(user, "<span class='warning'>You must resist in order to escape!</span>")
+	return
+
+/obj/item/clothing/head/mob_holder/container_resist()
+	if(isliving(loc))
+		var/mob/living/L = loc
+		visible_message("<span class='warning'>[src] escapes [L]!</span>")
+	release()
+
+/mob/living/proc/mob_pickup(mob/living/L)
+	var/obj/item/clothing/head/mob_holder/holder = generate_mob_holder()
+	if(!holder) return
+	drop_all_held_items()
+	L.put_in_hands(holder)
+	return
+
+/mob/living/proc/generate_mob_holder()
+	..()
+	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, (istext(can_be_held) ? can_be_held : ""), 'icons/mob/animals_held.dmi', 'icons/mob/animals_held_lh.dmi', 'icons/mob/animals_held_rh.dmi')
+	return holder
+
+/mob/living/simple_animal/drone/generate_mob_holder()
+	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, "[visualAppearence]_hat", null, null, null, TRUE)
+	return holder
+
+/mob/living/carbon/monkey/generate_mob_holder()
+	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, "monkey", 'icons/mob/animals_held.dmi', 'icons/mob/animals_held_lh.dmi', 'icons/mob/animals_held_rh.dmi', TRUE)
+	return holder
+
+/mob/living/simple_animal/mouse/generate_mob_holder()
+	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, (istext(can_be_held) ? can_be_held : ""), 'icons/mob/animals_held.dmi', 'icons/mob/animals_held_lh.dmi', 'icons/mob/animals_held_rh.dmi')
+	holder.w_class = 1
+	return holder
+
+/mob/living/proc/mob_try_pickup(mob/living/user)
+	if(!ishuman(user) || !src.Adjacent(user) || user.incapacitated() || !can_be_held)
+		return FALSE
+	if(user.get_active_held_item())
+		to_chat(user, "<span class='warning'>Your hands are full!</span>")
+		return FALSE
+	if(buckled)
+		to_chat(user, "<span class='warning'>[src] is buckled to something!</span>")
+		return FALSE
+	if(src == user)
+		to_chat(user, "<span class='warning'>You can't pick yourself up.</span>")
+		return FALSE
+	visible_message("<span class='warning'>[user] starts picking up [src].</span>", \
+					"<span class='userdanger'>[user] starts picking you up!</span>")
+	if(!do_after(user, 20, target = src))
+		return FALSE
+
+	if(user.get_active_held_item()||buckled)
+		return FALSE
+
+	visible_message("<span class='warning'>[user] picks up [src]!</span>", \
+					"<span class='userdanger'>[user] picks you up!</span>")
+	to_chat(user, "<span class='notice'>You pick [src] up.</span>")
+	mob_pickup(user)
+	return TRUE
+
+/mob/living/AltClick(mob/user)
```

Comment:
You definitely need to add it to the generic `examine()` proc. People won't know how to pick shit up.

```
if(can_be_held)
    to_chat(user, "<span class='notice'>They can be picked up with <b>Alt+Click</b>!</span>)
```

Reply:
Added it